scheduler.go

```diff
@@ -264,6 +275,10 @@ func (s *Scheduler) runAndReschedule(job *Job) error {
 }
 
 func (s *Scheduler) run(job *Job) error {
+	if s.sem != nil {
+		s.sem.Acquire(context.Background(), 1)
```

Comment:
Another question. Upon freezing here due to lack of weights, should this job's ryn be dropped? Otherwise they'll just get "throttled"